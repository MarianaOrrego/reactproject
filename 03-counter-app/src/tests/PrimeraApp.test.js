import { shallow} from 'enzyme';
import '@testing-library/jest-dom';
import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <PrimeraApp/>', () => {

    test('Debe de mostrar <PrimeraApp/> correctamente', () => {
        
        const saludo = 'Hola';
        const wrapper = shallow( <PrimeraApp saludo={ saludo }/> );

        expect( wrapper ).toMatchSnapshot();
    })
    
    test('Debe de mostrar el subtitulo enviado por props', () => {
        
        const saludo = 'Hola';
        const subtitulo = 'es noche';
        const wrapper = shallow( 
            <PrimeraApp 
                saludo={ saludo } 
                subtitulo={subtitulo}
            /> 
        );
        
        const textoParrafo = wrapper.find('p').text();
        console.log(textoParrafo);
        expect( textoParrafo ).toBe( subtitulo );
    })
    

//   test('Debe mostrar el mensaje "Hola"', () => {
    
//         const saludo = "Hola";
//         //renderizar funci√≥n
//         const { getByText } = render(<PrimeraApp saludo={ saludo } />);

//         expect( getByText(saludo) ).toBeInTheDocument();
//   })
  
})
