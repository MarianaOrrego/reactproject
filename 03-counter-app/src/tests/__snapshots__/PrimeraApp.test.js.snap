// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> Debe de mostrar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola
     
  </h1>
  <p>
    ya es martes
  </p>
</Fragment>
`;
